sys/kern/syscalls.c

```diff
@@ -310,24 +312,43 @@ static int sys_chdir(proc_t *p, chdir_args_t *args, register_t *res) {
   const char *u_path = args->path;
   char *path = kmalloc(M_TEMP, PATH_MAX, 0);
   size_t len = 0;
-  int error;
+  int error = 0;
 
   if ((error = copyinstr(u_path, path, PATH_MAX, &len)))
     goto end;
 
-  klog("chdir(\"%s\")", path);
-  error = ENOTSUP;
+  vnode_t *cwd;
+  error = vfs_namelookup(path, &cwd);
+  if (error)
+    goto end;
 
+  proc_self()->p_cwd = cwd;
 end:
   kfree(M_TEMP, path);
   return error;
 }
 
 static int sys_getcwd(proc_t *p, getcwd_args_t *args, register_t *res) {
-  __unused char *u_buf = args->buf;
-  __unused size_t len = args->len;
+  char *u_buf = args->buf;
+  size_t len = args->len;
 
-  return ENOTSUP;
+  char *path = kmalloc(M_TEMP, PATH_MAX, 0);
+  char *path_start = path + PATH_MAX;
```

Comment:
Why cannot you just pass pointer to `PATH_MAX` as second argument to `do_getcwd`?